pkg/skaffold/build/container_builder.go

```diff
@@ -222,6 +230,35 @@ watch:
 	return newTag, nil
 }
 
+func (cb *GoogleCloudBuilder) guessProjectID(artifact *v1alpha2.Artifact) (string, error) {
+	if cb.ProjectID != "" {
+		return cb.ProjectID, nil
+	}
+
+	ref, err := reference.ParseNormalizedNamed(artifact.ImageName)
+	if err != nil {
+		return "", errors.Wrap(err, "parsing image name for registry")
+	}
+
+	repoInfo, err := registry.ParseRepositoryInfo(ref)
+	if err != nil {
+		return "", err
+	}
+
+	index := repoInfo.Index
+	if !index.Official {
```

Comment:
@jonjohnsonjr is there a cleaner way to do this in go-containerregistry?

Reply:
It might be a little bit cleaner to use ggcr, but you're going to end up having to do string surgery regardless, so I wouldn't bother.

There's no notion of "Official" images on GCR, so this isn't super meaningful.

This also won't work for domain-scoped projects, though those are pretty long deprecated by now.